rafx-renderer/src/renderer.rs

```diff
@@ -241,21 +241,17 @@ impl Renderer {
     ) {
         let result = Self::try_create_render_job(&renderer, extract_resources, &presentable_frame);
 
-        let mut guard = renderer.inner.lock().unwrap();
-        let renderer_inner = &mut *guard;
         match result {
             Ok(prepared_frame) => {
-                if cfg!(all(feature = "no-render-thread")) {
-                    // NOTE(dvd): Run single threaded. Useful when trying to track # of global memory allocations.
-                    let _ = prepared_frame.render_async(
-                        presentable_frame,
-                        &*guard.render_thread.render_resources().lock().unwrap(),
-                    );
-                } else {
-                    renderer_inner
-                        .render_thread
-                        .render(prepared_frame, presentable_frame);
-                }
+                //if cfg!(all(feature = "no-render-thread")) {
+                // NOTE(dvd): Run single threaded. Useful when trying to track # of global memory allocations.
+                let _ = prepared_frame.render_async(presentable_frame);
```

Comment:
I think we need a flag on APIs like "supports multithreading", and if it's not enabled, take the single-threaded path